### PR TITLE
Update @tanstack/react-query to 5.8.1

### PR DESCRIPTION
**Pull request by aadamw**
Update react-query to 5.8.1 version

## Related Issue

Closes #

### CODE REVIEW

File: apps/web/src/app/[locale]/(email)/waitlist/_components/waitlist-form.tsx [hunk 1]
```diff
@@ -136,11 +136,11 @@ export function WaitlistForm() {
 
           <Button
             className="wl-form-button group relative mt-6 w-full overflow-hidden rounded-xl px-[2px] py-[2px] font-bold transition-shadow duration-300 hover:shadow-[0_0.5rem_2rem_-0.75rem_#3178c6] dark:hover:shadow-[0_0.5rem_2rem_-0.75rem_#5198f6]"
-            disabled={mutation.status === 'loading'}
+            disabled={mutation.status === 'pending'}
```
Comment (aadamw) [inline apps/web/src/app/[locale]/(email)/waitlist/_components/waitlist-form.tsx:139]: https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#status-loading-has-been-changed-to-status-pending-and-isloading-has-been-changed-to-ispending-and-isinitialloading-has-now-been-renamed-to-isloading

File: apps/admin/src/app/providers.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ interface Props {
 const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
-      cacheTime: 0,
+      gcTime: 0,
```
Comment (aadamw) [inline apps/admin/src/app/providers.tsx:15]: https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#status-loading-has-been-changed-to-status-pending-and-isloading-has-been-changed-to-ispending-and-isinitialloading-has-now-been-renamed-to-isloading

File: apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx [hunk 3]
```diff
@@ -121,8 +121,9 @@ export function Comment({
     hasNextPage: hasMoreReplies,
     refetch,
   } = useInfiniteQuery({
+    initialPageParam: 0,
```
Comment (aadamw) [inline apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:124]: [initialPageParam](https://tanstack.com/query/v5/docs/react/guides/migrating-to-v5#infinite-queries-now-need-a-initialpageparam)

File: apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx [hunk 4]
```diff
@@ -173,8 +174,8 @@ export function Comment({
         });
       }
       setReplyText('');
-      queryClient.invalidateQueries(replyQueryKey);
-      queryClient.invalidateQueries(queryKey);
+      queryClient.invalidateQueries({ queryKey: replyQueryKey });
```
Comment (aadamw) [inline apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:177]: `InvalidatesQueries` now accepts object

**Comment by bautistaaa** [comment]
will merge after i close out the pending PR's

**Comment by bautistaaa** [comment]
@aadamw i merged the comments PR. potentially things for you to update but unsure

**Comment by aadamw** [comment]
@bautistaaa updated

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/index.tsx [hunk 5]
```diff
@@ -69,7 +69,6 @@ export function Solutions({ slug }: Props) {
     queryKey,
     queryFn: () =>
       getPaginatedSolutions({ slug, page, sortKey: sortKey.key, sortOrder: sortKey.order }),
-    keepPreviousData: true,
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/index.tsx:72]: is this not a thing anymore?

**Review by bautistaaa — APPROVED** [review]

File: apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/index.tsx (hunk 5 above)
Comment (aadamw) [reply apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/index.tsx:72]: Looks like I missed it https://tanstack.com/query/latest/docs/react/guides/migrating-to-v5#removed-keeppreviousdata-in-favor-of-placeholderdata-identity-function

**Pull request merged**